modules/client/src/allowlistVoting/client.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  AllowlistVoting as AllowlistVotingContract,
+  AllowlistVoting__factory,
+} from "@aragon/core-contracts-ethers";
+import { ClientCore } from "../client-common";
+import { AllowlistVotingContextPlugin } from "./context";
+import { AllowlistVotingDecoding } from "./internal/decoding";
+import { AllowlistVotingEncoding } from "./internal/encoding";
+import { AllowlistVotingEstimation } from "./internal/estimation";
+import { AllowlistVotingMethods } from "./internal/methods";
+
+export class AllowlistVoting extends ClientCore {
+  public pluginInstance: AllowlistVotingContract;
+  public methods: AllowlistVotingMethods;
+  public estimation: AllowlistVotingEstimation;
+  public encoding: AllowlistVotingEncoding;
+  public decoding: AllowlistVotingDecoding;
+
+  constructor(context: AllowlistVotingContextPlugin) {
+    super(context);
+    this.pluginInstance = AllowlistVoting__factory.connect(
+      context.pluginAddress,
+      this.web3.getConnectedSigner()
```

Comment:
If we are going to use this `pluginInstance` across the whole plugin maybe it makes sense here to do a `this.web3.ensureOnline()`